core/mondoo-okta-security.mql.yaml

```diff
@@ -1,379 +1,1985 @@
 
-          ```bash
-          cnspec run okta -c "okta.policies.signOn.where(_.rules.all(status == "ACTIVE")).all(_.rules.all(actions["signon"]["session"]["maxSessionIdleMinutes"] <= 120))" --organization <YOUR-ORG> --token <OKTA_TOKEN>
-          cnspec run okta -c "okta.policies.signOn.where(_.rules.all(status == "ACTIVE")).all(_.rules.all(actions["signon"]["session"]["maxSessionLifetimeMinutes"] <= 1440))" --organization <YOUR-ORG> --token <OKTA_TOKEN>
-          ```
+        ### Okta recommends
 
-        __Use `cnspec shell`__
+        Require at least one factor in every MFA enrollment policy.
 
-        1. Launch the cnspec shell:
+        ### Security impact
 
-          ```bash
-          cnspec shell okta ---organization <YOUR-ORG> --token <OKTA_TOKEN>
-          ```
+        High 
 
-        2. Run this query:
+        ### End-user impact
 
-          ```mql
-          okta.policies.signOn.where(_.rules.all(status == "ACTIVE")).all(_.rules.all(actions["signon"]["session"]["maxSessionIdleMinutes"] <= 120))
-          okta.policies.signOn.where(_.rules.all(status == "ACTIVE")).all(_.rules.all(actions["signon"]["session"]["maxSessionLifetimeMinutes"] <= 1440))
-          ```
-      remediation: "### Use the Okta Admin Console\n\n1. Log into the organization as an administrator.\n2. Select **Security > Global Session Policy**.\n3. Select the rule to modify (select the pencil icon).\n4. Under **Session management**, select **Set time limit (Recommended)** and specify one day or lower.\n5. Under **Expire session after user has been idle on Okta for**, specify two hours or lower.\n6. Select **Update rule**.\n"
-  - uid: mondoo-okta-security-okta-auth-openid-saml
-    title: Use OpenID or SAML authentication for app access
-    impact: 80
+        Low
+
+        If a factor is required as part of the MFA enrollment policy, end users must enroll in the factor before they can sign in to their org. Setup varies depending on the factor specified.
+    variants:
+      - uid: mondoo-okta-security-okta-mfa-access-runtime
+  - uid: mondoo-okta-security-okta-mfa-access-runtime
+    title: Require at least one factor in every MFA enrollment policy
+    impact: 100
+    filters: 
+      asset.platform == "okta-org"
     mql: |
-      okta.applications.all(_.signOnMode == "OPENID_CONNECT" || _.signOnMode == "SAML_2_0")
+      everyoneGroupId = okta.groups.where( profile['name'] == "Everyone" ).map(id)[0]
+      okta.policies.mfaEnroll.one( status == "ACTIVE" && conditions['people']['groups']['include'].contains(everyoneGroupId) )
+    docs:
+      remediation: 
+        - id: console
+          desc: |
+            ### Set a required factor in an MFA enrollment policy
+            1. In the Admin Console, go to **Security > Multifactor**. The **Factor Types** page appears.
+            2. Click **Factor Enrollment** to switch to factor enrollment policies and rules.
```

Comment:
```suggestion
            2. Select **Factor Enrollment** to switch to factor enrollment policies and rules.
```

Reply:
@tas50 these are not our steps. they are from Okta's docs. They are copy/pasted.